pkg/odo/cli/registry/update.go

```diff
@@ -0,0 +1,97 @@
+package registry
+
+import (
+	// Build-in packages
+	"fmt"
+
+	// Third-party packages
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubernetes/pkg/kubectl/util/templates"
+
+	// odo packages
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/odo/cli/ui"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+	"github.com/openshift/odo/pkg/preference"
+)
+
+const updateCommandName = "update"
+
+// "odo registry update" command description and examples
+var (
+	updateLongDesc = ktemplates.LongDesc(`Update devfile registry`)
+
+	updateExample = ktemplates.Examples(`# Update devfile registry
+	%[1]s cheregistry https://che-devfile-registry-update.openshift.io/
+	`)
+)
+
+// UpdateOptions encapsulates the options for the "odo registry update" command
+type UpdateOptions struct {
+	operation    string
+	registryName string
+	registryURL  string
+	forceFlag    bool
+}
+
+// NewUpdateOptions creates a new UpdateOptions instance
+func NewUpdateOptions() *UpdateOptions {
+	return &UpdateOptions{}
+}
+
+// Complete completes UpdateOptions after they've been created
+func (o *UpdateOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.operation = "update"
+	o.registryName = args[0]
+	o.registryURL = args[1]
+	return
+}
+
+// Validate validates the UpdateOptions based on completed values
+func (o *UpdateOptions) Validate() (err error) {
+	return
```

Comment:
Again, IMO, there should be URL validation here like I've requested for `add.go`.

Reply:
IMO, we should not ask for a confirmation. The validation step should be able to tell the user that the registry they've specified doesn't exist:
```
$ odo registry update myreg22 google.com
? Are you sure you want to update registry myreg22 Yes
 ✗  Failed to update registry: registry myreg22 doesn't exist
```

Reply:
As pointed out for `odo registry add`, should we check that the registry has a valid `index.json` file so that user doesn't erroneously enter an invalid devfile registry?

Reply:
> IMO, we should not ask for a confirmation. The validation step should be able to tell the user that the registry they've specified doesn't exist:
> 
> ```
> $ odo registry update myreg22 google.com
> ? Are you sure you want to update registry myreg22 Yes
>  ✗  Failed to update registry: registry myreg22 doesn't exist
> ```

+1 This validation needs to happen first before we ask for confirmation

Reply:
> As pointed out for `odo registry add`, should we check that the registry has a valid `index.json` file so that user doesn't erroneously enter an invalid devfile registry?

Early validation like this has also its downsides. It means that if there is a  temporary network error, or for example, you want a registry that is inside the corporate network and you need VPN to access it you won't be able to do it unless it is accessible in a time of running `odo registry add`. Not saying that this is a necessary bad thing, but just something to consider

Reply:
I agree with the downside about network glitch. But what benefit does a user have by simply adding a registry behind the VPN if they're not going to be able to use it without being connected to VPN?

I feel it's worth adding validation here and in the `add` command to avoid unnecessarily failures later. Plus, the tools built on top of odo would able to just use it and hence the messaging remains the same across all the tools (IDE plugins) because it comes from odo!

Reply:
Regarding validation: https://github.com/openshift/odo/pull/2940#discussion_r417400387

Reply:
I would say we absolutely need the validation, but we need to consider where to add the validation, IMO that depends on the complexity of the validation itself and how long the validation takes. Given the registry validation is complicated and take pretty long time + we still goes over the same "iteration" when we use the registry + network glitch Tomas mentions, it would be better we add the validation on using the registry instead of configuring registry, it won't affect user perspective as finally users should use the registry and we gate there.

Reply:
We can add some early validation like checking if the URL format actually makes sense to avoid errors like @dharmit mentioned in https://github.com/openshift/odo/pull/2940#discussion_r417075309
Later we assume that the registry URL is http/https anyway, this check would make sense here. But I would avoid any additional checks that require a network, like if URL is actually pointing to the valid registry for reasons I described above. 

I would model this after Linux package managers. They allow you to add repo even if it is not accessible (you can do `yum-config-manager --add-repo` or `add-apt-repository` with URL that is not accessible, they just validate url format) but later they show unreachable repo warnings when the repo is not accessible (during `yum update` `apt update`).

Reply:
@kadel That's a good point! I think separate the validation is best option for us, we can both validate format error on registry configuring and validate registry error on registry usage, and I think it benefits for use case and product performance.

Reply:
> IMO, we should not ask for a confirmation. The validation step should be able to tell the user that the registry they've specified doesn't exist:
> 
> ```
> $ odo registry update myreg22 google.com
> ? Are you sure you want to update registry myreg22 Yes
>  ✗  Failed to update registry: registry myreg22 doesn't exist
> ```

Good catch! The confirmation should happens after the validation.

Reply:
> @kadel That's a good point! I think separate the validation is best option for us, we can both validate format error on registry configuring and validate registry error on registry usage, and I think it benefits for both use case and product performance.

+1. Separate validations makes lot of sense. :+1: